tests/Database/DatabaseEloquentBuilderTest.php

```diff
@@ -506,7 +506,7 @@ public function testHasWithContraintsAndJoinAndHavingInSubquery()
         $builder = $model->where('bar', 'baz');
         $builder->whereHas('foo', function ($q) {
             $q->join('quuuux', function ($j) {
-               $j->on('quuuuux', '=', 'quuuuuux', 'and', true);
+               $j->where('quuuuux', '=', 'quuuuuux');
```

Comment:
The last param (true) was removed from the ``on`` clause since it no longer makes sense.